Technomancy/tower.ts

```diff
@@ -197,49 +315,158 @@
+                   + tower_adj_b[adventure_data["tower_floor"] % tower_adj_b.length]
+                   + tower_adj_c[adventure_data["tower_floor"] % tower_adj_c.length]
+                   + tower_noun[adventure_data["tower_floor"] % tower_noun.length];
+        }
+        get text() {
+            return tower_rooms[adventure_data["tower_floor"] % tower_rooms.length];
+        }
+        "reward_text": "nothing but boasting rights",
+        reward: function () { }
+    },
+    { /* Boss Final, for each tower. */
+        "boss": "The final tower guardian",
+        "text": "It bounces around the room like a demented ferret, hissing and snarling as it goes. ",
+        "reward_text": "a new bigger shinier tower; oh and cheaper essence",
         reward: function () { }
     },
 ]
 
+let tower_ascension_growth=4;
 let grinding_level = 1;
 
+/* Adjectives and nouns to describe the boss.  Please ensure that the lengths of each set are mutually coprime with all the others. */
+/* Also, adjectives must end with a space or be totally empty. */
+let tower_adj_a = [ "beautiful ", "ugly ", "", "strange ", "corrupt ", "quaint ", "cute ", ""];
+let tower_adj_b = [ "big ", "small ", "baby ", "huge ", ""];
+let tower_adj_c = [ "green ", "", "blue ", "purple ", "red ", "striped ", "spotty " ];
+let tower_noun = ["fish", "frog", "triffid", "dog", "cat", "dove", "eagle", "goat", "golem"];
+/* Room descriptions.  Want to be a multiple of noun length so that the rooms match the noun. */
+let tower_rooms = ["It's just swimming around, looking angry. ",
+                   "It's hopping mad at you. ",
+                   "It's nearly filling the whole room, waiting to strike. ",
+                   "It looks like it might just sit if you told it to firmly.  Maybe not. ",
+                   "It gives you that look that says it knows that you're just a peasant in its eyes. ",
+                   "It flutters around looking innocent.  Maybe it is, maybe it isn't. ",
+                   "It soars round and round your head just waiting for you to take another step. ",
+                   "It nibbles the vegetation while watching you carefully. ",
+                   "It stands guard so still.  Will it move if you try and walk past? "];
+
+function tower_ascension_scale( initial, min, round ) {
+    if (round) {
+        Math.round(Math.max(initial / (1 + adventure_data["tower_ascension"]), min));
+    } else {
+        return Math.max(initial / (1 + adventure_data["tower_ascension"]), min);
+    }
+}
+
+/* Number of floors before final boss.
+function tower_height() {
+    return 30 + (adventure_data["tower_ascension"] * tower_ascension_growth);
+}
+
+function tower_boss_ascension_scale () {
+    var asc=adventure_data["tower_ascension"];
+    if (asc < 2) {
+        return 1;
+    }
+    var a=(1 + Math.sqrt(5))/2;
+    var b=(1 - Math.sqrt(5))/2;
+    return Math.round((Math.pow(a,adventure_data["tower_ascension"])
```

Comment:
Fun fact: This formula is actually exact, except that floating point operations can make it slightly off. But in actual math, it's exact.